### PR TITLE
GetState should call ads-serve endpoint

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/13213

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [x] New files have MPL-2.0 license header.
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/geographic/subdivision/get_subdivision_url_request_builder.cc [hunk 1]
```diff
@@ -29,7 +29,7 @@ UrlRequestPtr GetSubdivisionUrlRequestBuilder::Build() {
 ///////////////////////////////////////////////////////////////////////////////
 
 std::string GetSubdivisionUrlRequestBuilder::BuildUrl() const {
-  return base::StringPrintf("%s/v5/getstate", server::GetHost().c_str());
+  return base::StringPrintf("%s/v5/getstate", serve::server::GetHost().c_str());
```
Comment (amirsaber) [inline vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/geographic/subdivision/get_subdivision_url_request_builder.cc:32]: this should be /v1/getstate

**Review by amirsaber — CHANGES_REQUESTED** [review]
small fix is required

File: vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/geographic/subdivision/get_subdivision_url_request_builder.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/geographic/subdivision/get_subdivision_url_request_builder.cc:32]: We have always used v5, but checking with amir this worked due to a wildcard. Changing to v1

**Review by amirsaber — APPROVED** [review]

File: vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/geographic/subdivision/get_subdivision_url_request_builder.cc [hunk 2]
```diff
@@ -29,7 +29,7 @@ UrlRequestPtr GetSubdivisionUrlRequestBuilder::Build() {
 ///////////////////////////////////////////////////////////////////////////////
 
 std::string GetSubdivisionUrlRequestBuilder::BuildUrl() const {
-  return base::StringPrintf("%s/v5/getstate", server::GetHost().c_str());
+  return base::StringPrintf("%s/v1/getstate", serve::server::GetHost().c_str());
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/geographic/subdivision/get_subdivision_url_request_builder.cc:32]: @tmancey ticket states "Expected result: v5/getstate end-point should success and provide the regions subdivision (United States)", are we intending to serve from `v1`?

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/geographic/subdivision/get_subdivision_url_request_builder.cc:32]: @moritzhaller Please see https://github.com/brave/brave-core/pull/7453#discussion_r544127704, thanks

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/geographic/subdivision/get_subdivision_url_request_builder.cc:32]: Ah yes, good spot sir

**Review by moritzhaller — APPROVED** [review]
LGTM!

**Comment by tmancey** [comment]
CI failed for unrelates reasons

**Pull request merged**